crypto/bio/bio.c

```diff
@@ -240,7 +240,9 @@ int BIO_write_all(BIO *bio, const void *data, size_t len) {
   const uint8_t *data_u8 = data;
   while (len > 0) {
     int ret = BIO_write(bio, data_u8, len > INT_MAX ? INT_MAX : (int)len);
+    assert(ret <= (int)len);
```

Comment:
NP: The current assert would fail when `len > INT_MAX`.
```
    const int write_len = ((len > INT_MAX) ? INT_MAX : (int)len);
    int ret = BIO_write(bio, data_u8, write_len);
    assert(ret <= write_len);
```

Reply:
Would it? If `len > INT_MAX` was true, `BIO_write` would return a maximum of `INT_MAX` bytes if all were written. 
 In my mind it's `ret <= len < INT_MAX`.
I don't like the existing `int` cast, so I'll take your change though.

Reply:
ahh that cast behavior was the logic I was overlooking. Thanks for clearing things up!